tests/end-to-end/director/labeling_api_test.go

```diff
@@ -563,3 +565,173 @@ func TestDeleteDefaultValueInScenariosLabelDefinition(t *testing.T) {
+	applicationPage = ApplicationPageExt{}
+	err = tc.RunQuery(ctx, applicationRequest, &applicationPage)
+	require.NoError(t, err)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, applicationPage)
+	assert.Equal(t, applicationPage.TotalCount, 1)
+	assert.Contains(t, applicationPage.Data[0].Labels, labelKeyBar)
+	assert.Equal(t, applicationPage.Data[0].Labels[labelKeyBar], labelValueBar)
+	saveQueryInExamples(t, applicationRequest.Query(), "query applications with label filter")
+}
+
+func TestListLabelDefinitions(t *testing.T) {
+	//GIVEN
+	tenantID := uuid.New().String()
+	ctx := context.TODO()
+	firstSchema := map[string]interface{}{
+		"test": "test",
+	}
+	firstLabelDefinition := createLabelDefinitionWithinTenant(t, ctx, "first", firstSchema, tenantID)
+	defer deleteLabelDefinitionWithinTenant(t, ctx, firstLabelDefinition.Key, false, tenantID)
+
+	secondSchema := map[string]interface{}{
+		"test": "test",
+	}
+	secondLabelDefinition := createLabelDefinitionWithinTenant(t, ctx, "second", secondSchema, tenantID)
+	defer deleteLabelDefinitionWithinTenant(t, ctx, secondLabelDefinition.Key, false, tenantID)
+
+	//WHEN
+	labelDefinitions, err := listLabelDefinitionsWithinTenant(t, ctx, tenantID)
+
+	//THEN
+	require.NoError(t, err)
+	require.Len(t, labelDefinitions, 2)
+	assert.Contains(t, labelDefinitions, firstLabelDefinition)
+	assert.Contains(t, labelDefinitions, secondLabelDefinition)
+}
+
+func TestDeletingLastScenarioForApplication_ShouldFail(t *testing.T) {
+	//GIVEN
+	ctx := context.TODO()
+	tenant := uuid.New().String()
+	name := "test-deleting-last-scenario-for-application-should-fail"
+	scenarios := []string{"DEFAULT", "Christmas", "New Year"}
+
+	scenarioSchema := map[string]interface{}{
+		"type":        "array",
+		"minItems":    1,
+		"uniqueItems": true,
+		"items": map[string]interface{}{
+			"type": "string",
+			"enum": scenarios,
+		},
+	}
+	var schema interface{} = scenarioSchema
+
+	createLabelDefinitionWithinTenant(t, ctx, scenariosLabel, schema, tenant)
+
+	appInput := graphql.ApplicationInput{Name: name,
+		Labels: &graphql.Labels{
+			scenariosLabel: []string{"Christmas", "New Year"},
+		}}
+
+	application := createApplicationFromInputWithinTenant(t, ctx, appInput, tenant)
+	require.NotEmpty(t, application.ID)
+	defer deleteApplicationInTenant(t, application.ID, tenant)
+
+	//WHEN
+	appLabelRequest := fixSetApplicationLabelRequest(application.ID, scenariosLabel, []string{"Christmas"})
+	appLabelRequest.Header["Tenant"] = []string{tenant}
+	require.NoError(t, tc.RunQuery(ctx, appLabelRequest, nil))
+
+	//remove last label
+	appLabelRequest = fixSetApplicationLabelRequest(application.ID, scenariosLabel, []string{""})
+	appLabelRequest.Header["Tenant"] = []string{tenant}
+	err := tc.RunQuery(ctx, appLabelRequest, nil)
+
+	//THEN
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), `must be one of the following: "DEFAULT", "Christmas", "New Year"`)
+}
+
+func TestCreateRuntimeWithoutLabels(t *testing.T) {
```

Comment:
Shouldn't it be in Runtime test file? Or maybe even we test it already, as we check if runtime has correct properties (for example, no scenarios label if not provided)

Reply:
All runtimes in `runtime_api_test.go` contains Labels.
Moved this to the `runtime_api_test.go`